src/screens/login/index.tsx

```diff
@@ -8,58 +8,108 @@ import { login, registration } from '../../api/userAPI';
 import Background from '../../components/background/Background';
 import Title from '../../components/title/Title';
 
+import { ERROR_MESSAGE } from '../../../errors/errors';
 import { Icons } from '../../assets/Icons';
 import { Colors } from '../../constants/colors';
 import { TemplateType } from '../../constants/templateType';
 import { StackParams, UserType } from '../../types/types';
 
-import { ButtonText, LoginFrame, LoginInput, SignInButton, SignUpButton } from './styles';
+import { ButtonText, ErrorLogin, ErrorWindow,LoginFrame, LoginInput, SignInButton, SignUpButton, TextError } from './styles';
 
 const Login: FC = () => {
   const navigation = useNavigation<NativeStackNavigationProp<StackParams>>();
   const [email, setEmail] = useState<string | undefined>();
   const [password, setPassword] =useState<string | undefined>();
   const [user, setUser] = useState<UserType | undefined>();
+  const [isError, setIsError] = useState<boolean | undefined>(false );
+  const [errorMessage, setErrorMessage] = useState<string | undefined>();
 
   const signIn = async (): Promise<void> => {
     const userData = await login(email, password);
-    if (userData) {
-      setUser(userData);
+    if (typeof userData === 'number') {
+      setIsError(true);
+      setErrorMessage(ERROR_MESSAGE[userData]);
+    }
+    if ((userData as UserType).email) {
+      setIsError(false);
+      setUser((userData as UserType));
       navigation.navigate('Main');
     }
   };
 
   const signUp = async (): Promise<void> => {
     const userData = await registration(email, password);
-    if (userData) {
-      setUser(userData);
+    if (typeof userData === 'number') {
+      setIsError(true);
+      setErrorMessage(ERROR_MESSAGE[userData]);
+    }
+    if ((userData as UserType).email) {
+      setIsError(false);
+      setUser((userData as UserType));
+      navigation.navigate('Main');
     }
   };
 
+  const errorWindow = () => {
+    return(
+      <ErrorWindow>
+        <TextError>{errorMessage}</TextError>
+      </ErrorWindow>
+    );
+  };
+
+  const renderLogin = () => {
+    return isError ?
```

Comment:
write:
```
const Input = isError ? ErrorLogin : LoginInput;
return (
  <>
    <Input ... />
  </>
)
```

Reply:
when you see so much code duplication try to think whether there is a way to get rid of it